### PR TITLE
[addons] add next step on CAddonInfo to become independent

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description

This change all values on CAddonInfo to private and access
them over functions.

Main reason for them is to make the class more safe for next changes
where CAddonInfo is no longer in CAddon, but globally as replacement
for cpluff.

Set of this value becomes in future mainly done from the class constructor
and no more by external parts.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

**Review by ksooo — COMMENTED** [review]
Some suggestions and questions...

File: xbmc/addons/Addon.h [hunk 1]
```diff
@@ -56,38 +56,38 @@ class CAddon : public IAddon
   explicit CAddon(CAddonInfo addonInfo);
   virtual ~CAddon() {}
 
-  TYPE Type() const override { return m_addonInfo.type; }
+  TYPE Type() const override { return m_addonInfo.MainType(); }
```
Comment (tamland) [inline xbmc/addons/Addon.h]: In the end though there really should be no such thing as an 'addon type', but only addons with N possible extension points (which can have a type). So assuming this is only temporary for legacy code I guess 'MainType' works as to making it clearer..

File: xbmc/addons/AddonInfo.h [hunk 2]
```diff
@@ -85,13 +85,240 @@ namespace ADDON
...
 
     /*!
+     * @brief To ask generated class is usable and all needed parts are set.
+     *
+     * @return True is OK and usable, otherwise false.
+     *
+     * @note This function is only be related to constructor's who read the
+     * addon xml part.
+     */
+    bool IsUsable() const { return m_usable; }
```
Comment (tamland) [inline xbmc/addons/AddonInfo.h]: This doesn't belong to this commit

File: xbmc/addons/AddonInfo.h [hunk 3]
```diff
@@ -85,13 +85,240 @@ namespace ADDON
...
+
+    /*!
+     * @brief Parts used from ADDON::CAddonDatabase
+     */
+    //@{
+    void SetInstallDate(CDateTime installDate) { m_installDate = installDate; }
+    void SetLastUpdated(CDateTime lastUpdated) { m_lastUpdated = lastUpdated; }
+    void SetLastUsed(CDateTime lastUsed) { m_lastUsed = lastUsed; }
+    void SetOrigin(std::string origin) { m_origin = std::move(origin); }
+    //@}
```
Comment (tamland) [inline xbmc/addons/AddonInfo.h]: Same here: not used anywhere. And definitely not used by CAddonDatabase.

File: xbmc/addons/AddonInfo.h [hunk 4]
```diff
@@ -85,13 +85,240 @@ namespace ADDON
...
+
+    /*!
+     * @brief Get the ID from add-on
+     *
+     * The ID is defined in addon.xml but should also equal with add-on path!
```
Comment (tamland) [inline xbmc/addons/AddonInfo.h]: This isn't correct. There is no such requirement.

File: xbmc/addons/AddonInfo.h [hunk 5]
```diff
@@ -85,13 +85,240 @@ namespace ADDON
...
+     *
+     * @return Add-on name
+     */
+    const std::string& Name() const { return m_name; }
+
+    /*!
+     * @brief Licence text of add-on
+     *
+     * @return the complete licence text defined on addon.xml
+     */
```
Comment (tamland) [inline xbmc/addons/AddonInfo.h]: I find all these getter 'docstrings' very superficial.. It's completely obvious to everyone what all these methods does.

File: xbmc/addons/AddonInfo.h [hunk 6]
```diff
@@ -85,13 +85,240 @@ namespace ADDON
...
+     * @brief To get broken parts
+     *
+     * @todo make description more detailed
+     */
+    const std::string& Broken() const { return m_broken; }
+
+    /*!
+     * @brief To get install date of add-on
+     *
+     * @note this is only be set from add-on browser and not always present!
```
Comment (tamland) [inline xbmc/addons/AddonInfo.h]: That's not true. It should be set everywhere except when getting a single addon from addon manager. (And that is just a workaround due to some parts calling GetAddon way too much)

File: xbmc/addons/AddonBuilder.h [hunk 7]
```diff
@@ -30,36 +30,36 @@ class CAddonBuilder
...
+  void SetInstallDate(CDateTime installDate) { m_addonInfo.m_installDate = installDate; }
+  void SetLastUpdated(CDateTime lastUpdated) { m_addonInfo.m_lastUpdated = lastUpdated; }
+  void SetLastUsed(CDateTime lastUsed) { m_addonInfo.m_lastUsed = lastUsed; }
+  void SetOrigin(std::string origin) { m_addonInfo.m_origin = std::move(origin); }
+  void SetPackageSize(uint64_t size) { m_addonInfo.m_packageSize = size; }
 
-  const std::string& GetId() const { return m_addonInfo.id; }
-  const AddonVersion& GetVersion() const { return m_addonInfo.version; }
+  const std::string& GetId() const { return m_addonInfo.m_id; }
+  const AddonVersion& GetVersion() const { return m_addonInfo.m_version; }
```
Comment (AlwinEsch) [inline xbmc/addons/AddonBuilder.h]: The add of `friend class CAddonBuilder` is only temporarily to allow them to set the values in CAddonInfo, on end create CAddonInfo the values itself by his constructor.

**Review by tamland — CHANGES_REQUESTED** [review]

File: xbmc/addons/AddonInfo.h [hunk 8]
```diff
@@ -101,31 +328,40 @@ namespace ADDON
...
-    InfoMap extrainfo;
-    CDateTime installDate;
-    CDateTime lastUpdated;
-    CDateTime lastUsed;
-    std::string origin;
-    uint64_t packageSize;
+    InfoMap extrainfo; // defined here, but removed in future with cpluff
+
+  private:
+    friend class ADDON::CAddonBuilder;
```
Comment (AlwinEsch) [inline xbmc/addons/AddonInfo.h]: CAddonBuilder is the only class who currently need to set the values on CAddonInfo and only temporarily, the builder becomes on next steps complete removed. See also my other comment,

File: xbmc/addons/AddonInfo.h (hunk 2 above)
Comment (AlwinEsch) [inline xbmc/addons/AddonInfo.h]: Can remove it now, but becomes on next commits used the know class construction with read of addon.xml was OK.

File: xbmc/addons/AddonInfo.h (hunk 8 above)
Comment (ksooo) [inline xbmc/addons/AddonInfo.h]: Okay, then.

File: xbmc/addons/AddonBuilder.h (hunk 7 above)
Comment (ksooo) [inline xbmc/addons/AddonBuilder.h]: Okay.

File: xbmc/addons/AddonInfo.h (hunk 3 above)
Comment (AlwinEsch) [inline xbmc/addons/AddonInfo.h]: Also a part who becomes in future used from CAddonDatabase. Can remove it and add later.

File: xbmc/addons/AddonInfo.h (hunk 5 above)
Comment (AlwinEsch) [inline xbmc/addons/AddonInfo.h]: Better remove the docs, for me also not really needed?

Thought only that maybe wanted for there.

Comment (ksooo) [inline xbmc/addons/AddonInfo.h]: What @tamland said. Remove it. Does  not really add any value.

**Review by garbear — APPROVED** [review]

**Review by ksooo — APPROVED** [review]

**Comment by ksooo** [comment]
Let's handle documentation issues in a separate PR, once all code changes are in, like agreed before.

**Comment by AlwinEsch** [comment]
Have removed the comments, but want to leave the "Parts used from ADDON::CAddonDatabase" in. Becomes used on next commits and want them to hold small to see the critical parts.

**Comment by ksooo** [comment]
Fine by me. Merge it!

**Comment by AlwinEsch** [comment]
jenkins build this please

**Pull request merged**